OutOfSchool/OutOfSchool.AuthServer.Tests/EmailSender/EmailSenderJobTests.cs

```diff
@@ -0,0 +1,157 @@
+
+        // Assert
+        _mockSendGridClient.Verify(
+            client => client.SendEmailAsync(It.IsAny<SendGridMessage>(), It.IsAny<CancellationToken>()),
+            Times.Never);
+    }
+
+    [Test]
+    public async Task Execute_WithValidEmail_ShouldSendEmail()
+    {
+        // Arrange
+        _mockEmailOptions.Setup(options => options.Value).Returns(new EmailOptions { Enabled = true });
+        var mockContext = new Mock<IJobExecutionContext>();
+        mockContext.Setup(context => context.JobDetail.JobDataMap)
+            .Returns(new JobDataMap
+            {
+                { EmailSenderStringConstants.Email, "test@example.com" },
+                { EmailSenderStringConstants.Subject, "Test Email" },
+                { EmailSenderStringConstants.HtmlContent, Convert.ToBase64String(Encoding.ASCII.GetBytes("<html><body><h1>Hello</h1></body></html>")) },
+                { EmailSenderStringConstants.PlainContent, Convert.ToBase64String(Encoding.ASCII.GetBytes("Hello")) },
+                { EmailSenderStringConstants.ExpirationTime, DateTimeOffset.Now.AddDays(1) }
+            });
+        _mockSendGridClient.Setup(client => client.SendEmailAsync(It.IsAny<SendGridMessage>(), It.IsAny<CancellationToken>()))
+            .ReturnsAsync(new Response(HttpStatusCode.OK, null, null));
+
+        // Act
+        await _emailSenderJob.Execute(mockContext.Object);
+
+        // Assert
+        _mockSendGridClient.Verify(
+            client => client.SendEmailAsync(It.IsAny<SendGridMessage>(), It.IsAny<CancellationToken>()),
+            Times.Once);
+    }
+
+    [Test]
+    public void Execute_WithRateLimitExceeded_ThrowsJobExecutionException()
+    {
+        // Arrange
+        _mockEmailOptions.Setup(options => options.Value).Returns(new EmailOptions { Enabled = true });
+        var mockContext = new Mock<IJobExecutionContext>();
+        mockContext.Setup(context => context.JobDetail.JobDataMap)
+            .Returns(new JobDataMap
+            {
+                { EmailSenderStringConstants.Email, "test@example.com" },
+                { EmailSenderStringConstants.Subject, "Test Email" },
+                { EmailSenderStringConstants.HtmlContent, Convert.ToBase64String(Encoding.ASCII.GetBytes("<html><body><h1>Hello</h1></body></html>")) },
+                { EmailSenderStringConstants.PlainContent, Convert.ToBase64String(Encoding.ASCII.GetBytes("Hello")) },
+                { EmailSenderStringConstants.ExpirationTime, DateTimeOffset.Now.AddDays(1) }
+            });
+        _mockSendGridClient.Setup(client => client.SendEmailAsync(It.IsAny<SendGridMessage>(), It.IsAny<CancellationToken>()))
+            .ReturnsAsync(new Response(HttpStatusCode.TooManyRequests, null, null));
+
+        // Act & Assert
+        Assert.ThrowsAsync<JobExecutionException>(() => _emailSenderJob.Execute(mockContext.Object));
+    }
+
+    [Test]
+    public async Task Execute_WithSendGridError_ShouldThrowException()
```

Comment:
(Test name) The test name mismatch assert. Please fix it

Reply:
(I was bit inattentive while resolving this comment) Test name is still misleading. 

It states "ShouldNotSendEmail" but in assert you verify `_mockSendGridClient.Verify(... client.SendEmailAsync(...), Times.Once);`.

Maybe name should be something like `ShouldNotThrow` ?